### DISCUSSION TITLE
Modernisation of Carla - can we define the what, and how and where?

### DISCUSSION THREAD

**Discussion by shott92** [carla-simulator/carla#6237] [General]
Carla currently uses older versions of several critical components, and some now have very significant upgrade, or bring greater compatibility, such as the requirement for RX 7000 series GPU to have Linux Kernel 6 or newer for stable performance, as such it wont run properly on Ubuntu =<20.04.

**Ubuntu 22.04+** ships with GCC 11.2 and clang 1.13, it becomes rather challenging to build on this distro.
18.04 the currently supported distro is EOL in April, 2023 (2 months from writing this)
**RTX 7000 / RDNA3** graphics cards require mesa 23.0+, Linux Kernel 6.0+ and LLVM 15+, which can cause compatability issues with the other pre-requisites required for Carla to build on Linux
**Python 3.11** has now been released and is significantly faster than the 3.9 version supported by Carla 9.14
**UE5** is now launched, not only is it faster but the installation process on all platforms is much easier. it is also more performant, and visually more satisfying than UE4.26.
**Make 4.4** is recent and redily available the current requirement for make 3.81 from 2006, and is difficult to find and install correctly, on windows typically needs Chocolatey to install but often still fails.

**windows 8.1** SDK is no longer part of visual studio 2019 or 2022, Windows 11 is now out and again it is a challenge to install and run Carla on that platform.

I am wondering what the path to modernising Carla would be, and if it can be broken down into workstreams, for a new Carla version?
It is clear that updating carla, to support either ubuntu 22.04+, Python 3.11+ or UE5 would fundamentally break all backwards compatibility anyway, so can an official Carla Future fork to start progressing the development of this, while 0.9.14 can be maintained for current users on UE4.26?

Interested to hear the thoughts,
Thank you very much

**Comment by shott92** [comment]
**Unreal Engine Transition / migration:**
The documentation for the migration unreal engine from UE4.26 to UE5.x can be found here:
https://dev.epicgames.com/community/learning/paths/vQ/transitioning-from-ue4-to-ue5

**Python 3.11.x Porting from python 3.9:**
https://docs.python.org/3/whatsnew/3.10.html#porting-to-python-3-10
https://docs.python.org/3/whatsnew/3.11.html#porting-to-python-3-11

**build tools:**
**GCC**:
G++-7 is used for the compiling of Carla version 0.9.0 and perhaps earlier, G++-7 is last supported on Ubuntu 20.04 later versions require significant configuration:
Release and change notes can be found here:
https://gcc.gnu.org/

**LLVM and CLang**: from clang 10 to clang 15 (or maybe 17?)
Release notes can be found here it would seem the most significant changes occur between clang 10 and 11
https://releases.llvm.org/

**Boost**:
Current Dev version supports the latest Boost Version 1.81.0 
https://github.com/carla-simulator/carla/pull/6052/commits/ff2fd07866aba7216527a3fd57ba187fedd8298a Excellent work [Axel1092]

**Make**:
Make 3.81 cannot be easily found and installed on all Operating systems as such updating to a later more readily available release would be ideal it seem there are only backwards compatibility issues between ports to 3.82 and to 4.0, as such porting should be able to be done in two stages:
https://savannah.gnu.org/news/?group_id=71

Can we set-up a UE5 project on the projects tab and we can break these down into tasks for members to support the development?

**Comment by RickLamb** [comment]
Does anybody know offhand why we had to fork UE4?
( I'm guessing maybe the UE in docker changes which actually got integrated into mainline UE a while ago. )
